pkg/pipeline/util.go

```diff
@@ -0,0 +1,177 @@
+package pipeline
+
+import (
+	"bytes"
+	"log"
+	"math"
+	"text/template"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+type ResourceData struct{ *schema.ResourceData }
+
+func (d *ResourceData) getString(key string) string {
```

Comment:
I would love to see all this code common amongst the 4 projects in it's own repo C&P bugs me. 

Also, this code is a legacy remnant of the original RT atlassian code, which has been a headache

Reply:
Agree, we need to extract the common code among all the providers into a shared package.